tests/unit/test_orders.py

```diff
@@ -59,3 +61,66 @@ async def test_create_order_success(mock_http_post):
     # Note that create_orders() does not validate service responses.
     assert order.id == "lolwut"
     assert order.state == "queued"
+
+
+@pytest.mark.asyncio
+@patch("planet.clients.orders.ClientSession.get")
+async def test_wait_service_failure(mock_http_get):
+    """Get expected chained exception on fake service failure."""
+    mock_http_get.return_value.__aenter__.return_value.raise_for_status = Mock(
+        side_effect=RuntimeError("Boom! Service failure"))
+
+    # Order description endpoint URL construction requires a UUID id.
+    order = Order({"id": str(uuid.uuid4()), "state": "queued"})
+
+    client = OrdersClient(None)
+    with pytest.raises(OrderError) as excinfo:
+        await client.wait(order)
+
+    assert type(excinfo.value.__context__) == RuntimeError
+    assert str(excinfo.value.__context__) == "Boom! Service failure"
+
+
+@pytest.mark.asyncio
+@patch("planet.clients.orders.ClientSession.get")
+@patch("planet.clients.orders.POLL_SLEEP", 0.01)
+async def test_wait_timeout(mock_http_get):
+    """Get OrderError when max polling requests has been reached."""
+    # We script the service responses to always succeed, return "queued"
+    # the first time, and "running" infinitely thereafter.
+    mock_http_get.return_value.__aenter__.return_value.raise_for_status = Mock(
+        return_value=None)
+    mock_http_get.return_value.__aenter__.return_value.json.side_effect = itertools.chain(
+        [{
+            "state": "queued"
+        }], itertools.repeat({"state": "running"}))
```

Comment:
Reiterating a fact about this style of mocking: within the test function we don't really have an Orders service in play. No web requests are made. We're mocking the lower level HTTP client (from aiohttp) in a very limited way, programming only the responses to requests that we make within wait().

Reply:
Respx has more features, but in a lot of cases more than we need. The approach in this PR is more "pay only for what you eat". But also "cook everything from scratch" :smile: